### ISSUE TITLE
Context switching issues

### ISSUE THREAD

**Issue by DannyBen** [digitalocean/doctl#659]
I am not sure if I am doing it wrong, but the auth switching feature does not work as expected for me.

### 1. Possible bug: `DIGITALOCEAN_CONTEXT` not working

Following [these instructions][1], I understand that setting the `DIGITALOCEAN_CONTEXT` environment variable is the equivalent of using the `--context <context>` flag.

I expected this to work:

```
$ DIGITALOCEAN_CONTEXT=dannyb doctl account get
```

It does not. The account selected is **not** the one specified in the context.

If this environment variable is only intended to work with the `doctl auth init` command, than

1. it did not work there as well, and
2. it should work across commands. This will allow using some `.envrc` file that can be present on a per-directory basis, and set different contexts without having to type `--context <context>` all the time.

### 2. `doctl auth switch` confusion

Following the pattern of all other commands in the `doctl` CLI, I intuitively typed:

```
$ doctl auth context switch new-context
```

Of course, this does not work (but did not raise an error), as it expects `--context <context>` since it is a global flag.
Since this command is intended to do context switching, I would consider one of these options:

1. Allow this syntax to work: `doctl auth context switch new-context` (this would be my preference. The --context flag should NOT be accepted in this command).
2. Make the `--context` argument mandatory when running this command.
3. Raise an error when running without it, or when running with the extra (ignored) positional parameter, so we know we did something wrong.

[1]: https://github.com/digitalocean/doctl#logging-in-to-multiple-digitalocean-accounts

**Comment by bentranter** [comment]
Hey @DannyBen, thanks for raising this.

In the latest release, there's an issue around switching contexts via the `doctl auth switch --context` command that was solved in https://github.com/digitalocean/doctl/pull/651, but hasn't been included in a release. I'm going to cut a release for this now which should fix issues around that command as well as using the `DIGITALOCEAN_CONTEXT` flag.

As for your comments around the `--context` flag on the command, I agree that it's confusing – while I was manually testing the fix for 651, I made the same mistakes you did a number of times. Your suggestions on how to improve are great, so I'll consult with our team and see what we can do to improve, though my gut tells me it'll be suggestion 2 or 3.
